### PR TITLE
Adding typing to Postgres dialect file.

### PR DESCRIPTION
**Pull request by Gamrix**
Adding typing information for various parameters for Postgres types (in accordance to the docs).

This pull request is:

- [x] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [ ] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

File: lib/sqlalchemy/dialects/postgresql/types.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ class TIME(sqltypes.TIME):
 
     __visit_name__ = "TIME"
 
-    def __init__(self, timezone=False, precision=None):
+    def __init__(self, timezone=False, precision: Optional[int] = None):
```
Comment (CaselIT) [inline lib/sqlalchemy/dialects/postgresql/types.py:172]: Timezone should also be typed

File: lib/sqlalchemy/dialects/postgresql/types.py [hunk 2]
```diff
@@ -6,10 +6,10 @@
 # mypy: ignore-errors
```
Comment (CaselIT) [inline lib/sqlalchemy/dialects/postgresql/types.py:6]: This should be removed once the types are added

**Review by CaselIT — CHANGES_REQUESTED** [review]
Thanks for the effort, a few pointers

**Comment by CaselIT** [comment]
Thanks, let's see what the pipeline says

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision c7e39a219108f9e81ad22c008a664b62f09f9d5f of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change c7e39a219108f9e81ad22c008a664b62f09f9d5f: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4563

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4563 has been **merged**. Congratulations! :)

**Pull request closed without merging**